src/utils/rooms.js

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable import/prefer-default-export */
+
+export const isRoomAvailable = (room, when = Date.now()) => {
+  const { meetings } = room;
+
+  const actualValue = meetings.reduce((accumulator, actual) => {
+    const dateIni = new Date(actual.startTime).getTime();
+    const dateEnd = new Date(actual.endTime).getTime();
+    if (
+      dateIni <= when &&
+      (!accumulator.startTime ||
+        dateIni > new Date(accumulator.startTime).getTime())
+    ) {
+      if (dateEnd > when) {
+        return actual;
+      }
+    } else if (
+      !accumulator.startTime ||
+      dateIni < new Date(accumulator.startTime).getTime()
+    ) {
+      return actual;
+    }
+    return accumulator;
+  }, {});
```

Comment:
To improve readability, you can merge this two returns and do the if only in the time